web/api/wrapper.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+// WrapperCommand represents components necessary for OAuth2l request
+type WrapperCommand struct {
+	RequestType string
+	Args
+	Credential
+}
+
+// Args type used for unmarshalled JSON
+type Args map[string]interface{}
+
+// Credential type used for storing JSON-formatted credentials
+type Credential map[string]interface{}
+
+// Execute will capture output of OAuth2l CLI using command args
+func (wc WrapperCommand) Execute() (output string, err error) {
+	// combinedArgs used to represent command arguments in an array
+	args, ok := combinedArgs(wc)
+
+	// Check if credential is provided to include in oauth2l command
+	if wc.Credential != nil {
+		// Gets a file descriptor for a memory allocated credential file
+		descriptor, err := allocateMemFile(wc.Credential)
+
+		if err != nil {
+			return "", err
+		}
+
+		// Symlink path to memory file
+		path := getCredentialPath(descriptor)
```

Comment:
I will wrap this as:

path, err := getCredentialPath(wc.Credential)

Reply:
Just to confirm, I should add the `allocateMemFile` call in getCredentialPath?

Reply:
Yes 👍